From c4e4172d064ecf8822d03212c341adbf0a293c7c Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Fri, 29 Mar 2024 16:46:54 +0800
Subject: [PATCH 096/164] PHYTIUM: net/phytmac: Fixed the issue of pxe startup
 failure.

When network driver supports broadcast mode, no_broadcast bit of the
network config register needs to be set to 0.

Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I4721a61da370130dc6dce608448681b3fe8298d3

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/82a2fb0b2765a621ca2617c2471856264f909332
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac_v1.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_v1.c b/drivers/net/ethernet/phytium/phytmac_v1.c
index 2d9f67c82050..79df9a7f981f 100644
--- a/drivers/net/ethernet/phytium/phytmac_v1.c
+++ b/drivers/net/ethernet/phytium/phytmac_v1.c
@@ -357,8 +357,11 @@ static int phytmac_init_hw(struct phytmac *pdata)
 		config |= PHYTMAC_BIT(PROMISC);
 	if (pdata->ndev->features & NETIF_F_RXCSUM)
 		config |= PHYTMAC_BIT(RCO_EN);
-	if (!(pdata->ndev->flags & IFF_BROADCAST))
+	if (pdata->ndev->flags & IFF_BROADCAST)
+		config &= ~PHYTMAC_BIT(NO_BCAST);
+	else
 		config |= PHYTMAC_BIT(NO_BCAST);
+
 	/* pause enable */
 	config |= PHYTMAC_BIT(PAUSE_EN);
 	/* Rx Fcs remove */
-- 
2.47.0

